src/plone/api/relation.py

```diff
@@ -0,0 +1,230 @@
+    from plone.app.iterate.dexterity.relation import StagingRelationValue
+
+    # Plone 4 import error
+    from plone.app.linkintegrity.handlers import modifiedContent
+    from plone.app.linkintegrity.utils import referencedRelationship
+except ImportError:
+    ITERATE_RELATION_NAME = None
+    StagingRelationValue = None
+
+    # Plone 4 corrected paths
+    from plone.app.linkintegrity.handlers import modifiedDexterity as modifiedContent
+    from plone.app.linkintegrity.handlers import referencedRelationship
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_field_and_schema_for_fieldname(field_id, fti):
+    """Get field and its schema from a fti.
+    """
+    # Turn form.widgets.IDublinCore.title into title
+    field_id = field_id.split('.')[-1]
+    for schema in iterSchemataForType(fti):
+        field = schema.get(field_id, None)
+        if field is not None:
+            return (field, schema)
+
+
+@required_parameters('source', 'target', 'relationship')
+def create(source=None, target=None, relationship=None):
+    """Create a relation from source to target using zc.relation
+
+    For RelationChoice or RelationList it will add the relation as attribute.
+    Other relations they will only be added to the relation-catalog.
+
+    Copied from collective.relationhelpers link_objects.
+    """
+    if not IDexterityContent.providedBy(source):
+        raise InvalidParameterError('{} is no dexterity content'.format(source))
+
+    if not IDexterityContent.providedBy(target):
+        raise InvalidParameterError('{} is no dexterity content'.format(target))
+
+    if not isinstance(relationship, six.string_types):
+        raise InvalidParameterError('{} is no string'.format(relationship))
+
+    relation_catalog = getUtility(ICatalog)
+    intids = getUtility(IIntIds)
+    to_id = intids.getId(target)
+    from_id = intids.getId(source)
+    from_attribute = relationship
+
+    # Check if there is exactly this relation.
+    # If so remove it and create a fresh one.
+    query = {
+        'from_attribute': from_attribute,
+        'from_id': from_id,
+        'to_id': to_id,
+    }
+    for rel in relation_catalog.findRelations(query):
+        relation_catalog.unindex(rel)
+
+    if from_attribute == referencedRelationship:
+        # Don't mess with linkintegrity-relations!
+        # Refresh them by triggering this subscriber.
+        modifiedContent(source, None)
+        return
+
+    if ITERATE_RELATION_NAME is not None and from_attribute == ITERATE_RELATION_NAME:
+        # Iterate relations use a subclass of RelationValue
+        relation = StagingRelationValue(to_id)
+        event._setRelation(source, ITERATE_RELATION_NAME, relation)
+        return
+
+    fti = queryUtility(IDexterityFTI, name=source.portal_type)
+    if not fti:
+        logger.info(u'{} is no dexterity content'.format(source.portal_type))
+        return
+    field_and_schema = _get_field_and_schema_for_fieldname(from_attribute, fti)
```

Comment:
also DRY - controlpanel
also see my comments about FTI in controlpanel PR

Reply:
Comments there were hidden by now, but I found it:

> it is OK to use a portal_type string as FTI here, DX schema cache handles lookup transparent in high-performance way. So all calls of this function can save a getUtility call, which may speed up the whole process a bit and saves CPU cycles, time and CO2.

Done.